model_compression_toolkit/qunatizers_infrastructure/pytorch/quantize_wrapper.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================f
+
+
+from model_compression_toolkit.core.common.constants import FOUND_TORCH
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.qunatizers_infrastructure.common.node_quantization_dispatcher import \
+    NodeQuantizationDispatcher
+
+if FOUND_TORCH:
+    import torch
+    import torch.nn as nn
+
+    DISPATCHER = "dispatcher"
+    LAYER = "layer"
+    STEPS = "optimizer_step"
+    TRAINING = "training"
+
+
+    def _weight_name(name: str) -> str:
+        """Extracts the weight name from the full TensorFlow variable name.
+
+        For example, returns 'kernel' for 'dense_2/kernel:0'.
+
+        Args:
+          name: TensorFlow variable name.
+
+        Returns:
+          Extracted weight name.
+        """
+        return name.split(':')[0].split('/')[-1]
```

Comment:
This is duplicated with Keras quantize_wrapper.
If this is a convention for a weights name structure and we use this extraction frequently, maybe put it in some common location (create "weights_name_extractor" function or something) and use it throughout the code.

I believe that we're also using it in [qat/keras/quantization_facade.py](https://github.com/sony/model_optimization/blob/6d2831e4ce03d23ca0a0e173166a6b210e1af2b2/model_compression_toolkit/qat/keras/quantization_facade.py#L231)
and in [fakely_quant_keras_exporter.py](https://github.com/sony/model_optimization/blob/6d2831e4ce03d23ca0a0e173166a6b210e1af2b2/model_compression_toolkit/exporter/model_exporter/keras/fakely_quant_keras_exporter.py#L103)
so if you create such function use it there as well.

Reply:
this function is a copy from the tensorflow code. make sure it's needed for pytorch